.github/workflows/npm-audit.yml

```diff
@@ -0,0 +1,69 @@
+name: NPM Audit Fix
+
+on:
+  schedule:
+    - cron: "0 7 * * 1" # Weekly on Monday at 7 AM UTC
+  workflow_dispatch:
+
+jobs:
+  npm-audit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v5
+      
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20"
+          
+      - name: NPM install and audit fix
+        working-directory: src/Misc/expressionFunc/hashFiles
+        run: |
+          npm install
+          npm audit fix --force
```

Comment:
we probably still want to run `npm run all` after the `audit fix`?

Reply:
adjusted, i've adjusted it so it always commits the fixes, and mentions that there are build errors to be fixed if needed, what about having the complex workflow separately that can be manually run?